authors/apps/core/utils.py

```diff
@@ -0,0 +1,72 @@
+from datetime import datetime, timedelta
+import jwt
+from django.conf import settings
+from rest_framework import exceptions
+
+import os
+from sendgrid import SendGridAPIClient
+from sendgrid.helpers.mail import Mail
+
+class TokenHandler:
+    """This class contains the methods for creating custom
+    tokens for users"""
+
+    def create_verification_token(self, payload):
+        """
+        Create a JWT token to be sent in the verification
+        email url
+        """
+
+        if not isinstance(payload, dict):
+            raise TypeError('Payload must be a dictionary!')
+
+        token_expiry = datetime.now() + timedelta(hours=24)
+
+        try:
+            token = jwt.encode({
+                'email': payload['email'],
+                'exp': int(token_expiry.strftime('%s'))},
+                settings.SECRET_KEY, algorithm='HS256')
+            return token.decode('utf-8')
+
+        except KeyError:
+            return "Please provide email"
+
+    def validate_token(self, token):
```

Comment:
This function takes in a token and decodes it, there is a function that performs a similar action within the 'jwt_generator.py' module. wouldn't it be less repetitive to simply import that one instead?

Reply:
Good point, though the specific method you are suggesting I use wont work for my case because of the parameters the function takes. That is the reason I had to create my own custom method for token generation.

Reply:
I agree with @ipaullly that this method looks quite similar to the decode method in the jwt_generator.py and they seem to do the same thing - only difference is the exceptions. Is there a way you can reuse the method in the jwt_generator.py

Reply:
it makes sense if it is for a specific purpose

Reply:
@larryTheGeek kindly look into the changes made by @ipaullly on the same method. I think you can try work with the existing jwt_decode method in the jwt_generator.py

Reply:
> @larryTheGeek kindly look into the changes made by @ipaullly on the same method. I think you can try work with the existing jwt_decode method in the jwt_generator.py

Let me look into it

Reply:
> @larryTheGeek kindly look into the changes made by @ipaullly on the same method. I think you can try work with the existing jwt_decode method in the jwt_generator.py

I have found  a way to work with it kindly confirm